README.md

```diff
@@ -2,302 +2,708 @@
+  end
+
+  def initialize(user, input_data)
+    super input_data
+    @user = user
+  end
+
+  private
+  def run
+    @user.update params
+  end
 end
 ```
 
+### Schema-wide policies
+
+Sometimes it's useful to apply the same policy to all fields in a schema.
+
+For example, fields that are _required_ when creating a record might be optional when updating the same record (ie. _PATCH_ operations in APIs).
+
 ```ruby
-order_params = OrdersParams.new(page: 2, q: 'foobar')
-order_params[:page] # => 2
-order_params[:per_page] # => 30
-order_params.each{|key, value| ... }
+class UpdateUserForm < CreateUserForm
+  schema.policy(:declared)
+end
 ```
 
-## Nested structures
+This will prefix the `:declared` policy to all fields inherited from the parent class.
+This means that only fields whose keys are present in the input will be validated.
 
-You can also nest parameter definitions. This is useful if you need to model POST payloads, for example.
+Schemas with default policies can still define or re-define fields.
 
 ```ruby
-class AccountPayload
-  include Parametric::Params
-  param :status, 'Account status', default: 'pending', options: ['pending', 'active', 'cancelled']
-  param :users, 'Users in this account', multiple: true do
-    param :name, 'User name'
-    param :title, 'Job title', default: 'Employee'
-    param :email, 'User email', match: /\w+@\w+\.\w+/
-  end
-  param :owner, 'Owner user' do
-    param :name, 'User name'
-    param :email, 'User email', match: /\w+@\w+\.\w+/
+class UpdateUserForm < CreateUserForm
+  schema.policy(:declared) do
+    # Validation will only run if key exists
+    field(:age).type(:integer).present
   end
 end
 ```
 
-The example above expects a data structure like the following:
+### Ignoring fields defined in the parent class
+
+Sometimes you'll want a child class to inherit most fields from the parent, but ignoring some.
 
 ```ruby
-{
-  status: 'active',
-  users: [
-    {name: 'Joe Bloggs', email: 'joe@bloggs.com'},
-    {name: 'jane Bloggs', email: 'jane@bloggs.com', title: 'CEO'}
-  ],
-  owner: {
-    name: 'Olivia Owner',
-    email: 'olivia@owner.com'
-  }
-}
+class CreateUserForm
+  include Parametric::DSL
+
+  schema do
+    field(:uuid).present
+    field(:status).required.options(["inactive", "active"])
+    field(:name)
+  end
+end
 ```
 
-## Use cases
+The child class can use `ignore(*fields)` to ignore fields defined in the parent.
 
-### In Rails
+```ruby
+class UpdateUserForm < CreateUserForm
+  schema.ignore(:uuid, :status) do
+    # optionally add new fields here
+  end
+end
+```
+
+## Schema options
 
-You can use one-level param definitions in GET actions
+Another way of modifying inherited schemas is by passing options.
 
 ```ruby
-def index
-  @search = OrdersSearch.new(params)
-  @results = @search.results
+class CreateUserForm
+  include Parametric::DSL
+
+  schema(default_policy: :noop) do |opts|
+    field(:name).validate(opts[:default_policy).type(:string).required
```

Comment:
This is where you start to lose me. What really is `.opts` (the parent schema maybe?) and what is `default_policy`? (is it just a way of saying "inherit from this policy rather than the policy of the parent class"?)

Also, I don't get what `(opts[:default_policy).type(:string).required` is actually doing. Is it missing a square bracket? (eg. `(opts[:default_policy])`) although even if that is the case, wouldn't that be validating `:noop`?